### PR TITLE
Investigate withoutSync behavior under hashrouting enabled #1787

### LINKED ISSUE #1787: Investigate withoutSync behavior under hashrouting enabled
_withoutSync_ based navigation command seems to navigate to the path when _useHashRouting_ is set to **"true"**:
`LuigiClient.linkManager().withoutSync().navigate('path/to');`
Normally it shouldn't. Investigate the behavior and a possible fix.

### PR DESCRIPTION
**Pull request by legteodav**
Fixes #1787

### CODE REVIEW

**Review by stanleychh — APPROVED** [review]

File: core/src/services/routing.js [hunk 1]
```diff
@@ -58,6 +58,9 @@ class RoutingClass {
     }
 
     if (LuigiConfig.getConfigValue('routing.useHashRouting')) {
+      if (!navSync){
+        EventListenerHelpers.hashChangeWithoutSync=true;
```
Comment (ndricimrr) [inline core/src/services/routing.js:62]:
```suggestion
        EventListenerHelpers.hashChangeWithoutSync = true;
```

File: core/src/utilities/helpers/event-listener-helpers.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class EventListenerHelpersClass {
   constructor() {
     this.listeners = [];
+    this.hashChangeWithoutSync=false;
```
Comment (ndricimrr) [inline core/src/utilities/helpers/event-listener-helpers.js:4]:
```suggestion
    this.hashChangeWithoutSync = false;
```

File: core/src/utilities/helpers/routing-helpers.js [hunk 3]
```diff
@@ -135,7 +135,9 @@ class RoutingHelpersClass {
 
     if (hashRoutingActive) {
       return EventListenerHelpers.addEventListener('hashchange', event => {
-        callback(Routing.getHashPath(event.newURL));
+        const withoutSync=EventListenerHelpers.hashChangeWithoutSync;
+        EventListenerHelpers.hashChangeWithoutSync=false;
```
Comment (ndricimrr) [inline core/src/utilities/helpers/routing-helpers.js:139]:
```suggestion
        EventListenerHelpers.hashChangeWithoutSync = false;
```

**Comment by JohannesDoberer** [comment]
Jimi will investigate further scenarios

File: core/src/utilities/helpers/routing-helpers.js [hunk 4]
```diff
@@ -135,7 +135,9 @@ class RoutingHelpersClass {
 
     if (hashRoutingActive) {
       return EventListenerHelpers.addEventListener('hashchange', event => {
-        callback(Routing.getHashPath(event.newURL));
+        const withoutSync=EventListenerHelpers.hashChangeWithoutSync;
```
Comment (ndricimrr) [inline core/src/utilities/helpers/routing-helpers.js:138]:
```suggestion
        const withoutSync = EventListenerHelpers.hashChangeWithoutSync;
```

**Review by ndricimrr — APPROVED** [review]

**Pull request merged**